tket/src/Gate/Rotation.cpp

```diff
@@ -462,6 +462,25 @@ std::vector<double> tk1_angles_from_unitary(const Eigen::Matrix2cd &U) {
   return {a, b, c, p};
 }
 
+std::tuple<double, double, double> get_bloch_coordinate_from_state(
+    const Complex &a, const Complex &b) {
+  // We require the state to be normalised
+  double norm = a.real() * a.real() + a.imag() * a.imag() +
+                b.real() * b.real() + b.imag() * b.imag();
+  if (std::abs(norm - 1) > EPS) {
+    throw std::invalid_argument(
+        "Attempt to find the block sphere coordinates for an unnormalised "
```

Comment:
Could you update this error message to also explain why it failed please

Reply:
Is the updated message better? or do I still need to add something?